### PR TITLE
refactor: Unify logging

### PR DESCRIPTION
**Pull request by sidferreira**
Summary:
---------

Based on #95, replaced `console.(log|warn|error)` by `logger.(info|warn|error)`.
Used regular expressions but everything seems to be working as should.

Test Plan:
----------

All tests are passing, no other change was required.

### CODE REVIEW

File: packages/cli/src/generator/printRunInstructions.js [hunk 1]
```diff
@@ -22,19 +23,19 @@ function printRunInstructions(projectDir, projectName) {
...
     xcodeProjectPath
   );
-  console.log(chalk.white.bold('To run your app on iOS:'));
-  console.log(`   cd ${absoluteProjectDir}`);
-  console.log('   react-native run-ios');
-  console.log('   - or -');
-  console.log(`   Open ${relativeXcodeProjectPath} in Xcode`);
-  console.log('   Hit the Run button');
+  logger.info(chalk.white.bold('To run your app on iOS:'));
+  logger.info(`   cd ${absoluteProjectDir}`);
```
Comment (thymikee) [inline packages/cli/src/generator/printRunInstructions.js]: `logger.info` will prepend ` INFO ` to the message, we don't want it to be here. Can you fold it down to a single `logger.info` call with a nice template string inside?

File: packages/cli/src/logIOS/logIOS.js [hunk 2]
```diff
@@ -37,7 +38,7 @@ async function logIOS() {
 
   const device = findAvailableDevice(devices);
   if (device === undefined) {
-    console.log(chalk.red('No active iOS device found'));
+    logger.info(chalk.red('No active iOS device found'));
```
Comment (thymikee) [inline packages/cli/src/logIOS/logIOS.js]:
```suggestion
    logger.error('No active iOS device found');
```

File: packages/cli/src/runAndroid/runAndroid.js [hunk 3]
```diff
@@ -33,7 +34,7 @@ function checkAndroid(root) {
  */
 function runAndroid(argv: Array<string>, config: ContextT, args: Object) {
   if (!checkAndroid(args.root)) {
-    console.log(
+    logger.info(
       chalk.red(
```
Comment (thymikee) [inline packages/cli/src/runAndroid/runAndroid.js]: this should be `logger.error` without chalk

File: packages/cli/src/runAndroid/runAndroid.js [hunk 4]
```diff
@@ -47,14 +48,14 @@ function runAndroid(argv: Array<string>, config: ContextT, args: Object) {
 
   return isPackagerRunning(args.port).then(result => {
     if (result === 'running') {
-      console.log(chalk.bold('JS server already running.'));
+      logger.info(chalk.bold('JS server already running.'));
     } else if (result === 'unrecognized') {
-      console.warn(
+      logger.warn(
         chalk.yellow('JS server not recognized, continuing with build...')
```
Comment (thymikee) [inline packages/cli/src/runAndroid/runAndroid.js]:
```suggestion
       'JS server not recognized, continuing with build...'
```

**Comment by sidferreira** [comment]
```
Cannot call `logger.info` because array type [1] is incompatible with string [2] in array element.

   packages/cli/src/runAndroid/runAndroid.js:138:7
   138|       logger.info(devices);
              ^^^^^^^^^^^^^^^^^^^^

References:
   packages/cli/src/runAndroid/adb.js:37:24
    37| function getDevices(): Array<string> {
                               ^^^^^^^^^^^^^ [1]
   packages/cli/src/util/logger.js:10:34
    10| const info = (...messages: Array<string>) => {
                                         ^^^^^^ [2]
```

Should we change the type to accept arrays or actually split the parameter?

File: packages/cli/src/runAndroid/runAndroid.js [hunk 5]
```diff
@@ -47,14 +48,14 @@ function runAndroid(argv: Array<string>, config: ContextT, args: Object) {
 
   return isPackagerRunning(args.port).then(result => {
     if (result === 'running') {
-      console.log(chalk.bold('JS server already running.'));
+      logger.info(chalk.bold('JS server already running.'));
```
Comment (thymikee) [inline packages/cli/src/runAndroid/runAndroid.js]:
```suggestion
      logger.info('JS server already running.');
```

File: packages/cli/src/runAndroid/runAndroid.js [hunk 6]
```diff
@@ -101,7 +102,7 @@ function buildAndRun(args) {
         adbPath
       );
     }
-    console.log(chalk.red('Argument missing for parameter --deviceId'));
+    logger.info(chalk.red('Argument missing for parameter --deviceId'));
```
Comment (thymikee) [inline packages/cli/src/runAndroid/runAndroid.js]:
```suggestion
    logger.error('Argument missing for parameter --deviceId');
```

File: packages/cli/src/runAndroid/runAndroid.js [hunk 7]
```diff
@@ -47,14 +48,14 @@ function runAndroid(argv: Array<string>, config: ContextT, args: Object) {
...
+      logger.info(chalk.bold('JS server already running.'));
     } else if (result === 'unrecognized') {
-      console.warn(
+      logger.warn(
         chalk.yellow('JS server not recognized, continuing with build...')
       );
     } else {
       // result == 'not_running'
-      console.log(chalk.bold('Starting JS server...'));
+      logger.info(chalk.bold('Starting JS server...'));
```
Comment (thymikee) [inline packages/cli/src/runAndroid/runAndroid.js]:
```suggestion
      logger.info('Starting JS server...');
```

**Comment by sidferreira** [comment]
- [ ] log(chalk.red => error
- [ ] Remove all bold
- [ ] chalk.yellow => warn

File: packages/cli/src/runAndroid/runAndroid.js [hunk 8]
```diff
@@ -132,25 +133,25 @@ function runOnSpecificDevice(
         adbPath
       );
     } else {
-      console.log(`Could not find device with the id: "${args.deviceId}".`);
-      console.log('Choose one of the following:');
-      console.log(devices);
+      logger.info(`Could not find device with the id: "${args.deviceId}".`);
+      logger.info('Choose one of the following:');
+      logger.info(devices);
```
Comment (thymikee) [inline packages/cli/src/runAndroid/runAndroid.js]: please fold it into single `logger.error`

File: packages/cli/src/runAndroid/runAndroid.js [hunk 9]
```diff
@@ -132,25 +133,25 @@ function runOnSpecificDevice(
...
     }
   } else {
-    console.log('No Android devices connected.');
+    logger.info('No Android devices connected.');
```
Comment (thymikee) [inline packages/cli/src/runAndroid/runAndroid.js]:
```suggestion
    logger.error('No Android devices connected.');
```

File: packages/cli/src/runAndroid/runAndroid.js (hunk 7 above)
Comment (sidferreira) [reply packages/cli/src/runAndroid/runAndroid.js]: @thymikee remove all `chalk.bold`?

**Comment by thymikee** [comment]
>Should we change the type to accept arrays or actually split the parameter?

just spread the devices:
```js
logger.info(...devices);
```

**Comment by sidferreira** [comment]
```
module.exports = function copyMiddleware(req, res, next) {
  if (req.url === '/copy-to-clipboard') {
    const ret = copyToClipBoard(req.rawBody);
    if (!ret) {
      logger.warn(chalk.red('Copy button is not supported on this platform!'));
    }
    res.end('OK');
  } else {
    next();
  }
};
```

error or warn? @thymikee

**Comment by thymikee** [comment]
I'd go with regular `warn` for now :)

File: packages/cli/src/generator/printRunInstructions.js (hunk 1 above)
Comment (sidferreira) [reply packages/cli/src/generator/printRunInstructions.js]: Fixed

File: packages/cli/src/logIOS/logIOS.js (hunk 2 above)
Comment (sidferreira) [reply packages/cli/src/logIOS/logIOS.js]: Fixed

File: packages/cli/src/runAndroid/runAndroid.js (hunk 3 above)
Comment (sidferreira) [reply packages/cli/src/runAndroid/runAndroid.js]: Fixed

File: packages/cli/src/runAndroid/runAndroid.js (hunk 4 above)
Comment (thymikee) [reply packages/cli/src/runAndroid/runAndroid.js]: this (and similar) still need to be addressed

Comment (sidferreira) [reply packages/cli/src/runAndroid/runAndroid.js]: fixed

File: packages/cli/src/runAndroid/runAndroid.js (hunk 5 above)
Comment (sidferreira) [reply packages/cli/src/runAndroid/runAndroid.js]: fixed

File: packages/cli/src/runAndroid/runAndroid.js (hunk 6 above)
Comment (sidferreira) [reply packages/cli/src/runAndroid/runAndroid.js]: fixed

File: packages/cli/src/runAndroid/runAndroid.js (hunk 7 above)
Comment (sidferreira) [reply packages/cli/src/runAndroid/runAndroid.js]: fixed

File: packages/cli/src/runAndroid/runAndroid.js (hunk 8 above)
Comment (sidferreira) [reply packages/cli/src/runAndroid/runAndroid.js]: fixed

File: packages/cli/src/runAndroid/runAndroid.js (hunk 9 above)
Comment (sidferreira) [reply packages/cli/src/runAndroid/runAndroid.js]: fixed

File: packages/cli/src/server/util/debugger-ui/debuggerWorker.js [hunk 10]
```diff
@@ -18,11 +18,11 @@ onmessage = (function() {
     var hasWarned = false;
     return function() {
       // Wait until `YellowBox` gets initialized before displaying the warning.
-      if (hasWarned || console.warn.toString().includes('[native code]')) {
+      if (hasWarned || logger.warn.toString().includes('[native code]')) {
         return;
       }
       hasWarned = true;
-      console.warn(
+      logger.warn(
```
Comment (thymikee) [inline packages/cli/src/server/util/debugger-ui/debuggerWorker.js]: we don't want logger in `debugger-ui` because it runs in a browser

Comment (sidferreira) [reply packages/cli/src/server/util/debugger-ui/debuggerWorker.js]: fixed

File: packages/cli/src/runIOS/runIOS.js [hunk 11]
```diff
@@ -251,12 +252,12 @@ async function runOnDevice(
...
 
   if (iosDeployOutput.error) {
-    console.log('');
-    console.log('** INSTALLATION FAILED **');
-    console.log('Make sure you have ios-deploy installed globally.');
-    console.log('(e.g "npm install -g ios-deploy")');
+    logger.info('');
+    logger.info('** INSTALLATION FAILED **');
+    logger.info('Make sure you have ios-deploy installed globally.');
+    logger.info('(e.g "npm install -g ios-deploy")');
```
Comment (thymikee) [inline packages/cli/src/runIOS/runIOS.js]: can we fold this?

File: packages/cli/src/runAndroid/tryRunAdbReverse.js [hunk 12]
```diff
@@ -23,13 +23,13 @@ function tryRunAdbReverse(packagerPort: number | string, device: string) {
       adbArgs.unshift('-s', device);
     }
 
-    console.log(chalk.bold(`Running ${adbPath} ${adbArgs.join(' ')}`));
+    logger.info(chalk.bold(`Running ${adbPath} ${adbArgs.join(' ')}`));
```
Comment (thymikee) [inline packages/cli/src/runAndroid/tryRunAdbReverse.js]:
```suggestion
    logger.info(`Running ${adbPath} ${adbArgs.join(' ')}`);
```

**Comment by sidferreira** [comment]
```
react-native-cli/packages/cli/src/init/init.js
  53:3  error  Unused eslint-disable directive (no problems were reported from 'import/no-unresolved')

  // eslint-disable-next-line import/no-unresolved
  const reactNativePackageJson = require('react-native/package.json');
```

**Comment by thymikee** [comment]
>53:3  error  Unused eslint-disable directive (no problems were reported from 'import/no-unresolved')

Try rebasing/merging master, as after this PR https://github.com/react-native-community/react-native-cli/pull/144 we dropped `react-native` as a dev dependency

File: packages/cli/src/runIOS/runIOS.js [hunk 13]
```diff
@@ -82,11 +82,10 @@ function runIOS(_: Array<string>, ctx: ContextT, args: FlagsT) {
     }
     if (devices && devices.length > 0) {
       // $FlowIssue: args.device is defined in this context
-      logger.info(`Could not find device with the name: "${args.device}".`);
-      logger.info('Choose one of the following:');
-      printFoundDevices(devices);
+      logger.error(`Could not find device with the name: "${args.device}".
+Choose one of the following:${printFoundDevices(devices)}`);
```
Comment (thymikee) [inline packages/cli/src/runIOS/runIOS.js]: `printFoundDevices` outputs logs by itself, so this is wrong

Comment (sidferreira) [reply packages/cli/src/runIOS/runIOS.js]: @thymikee I changed it to be a list.
There's another place which the output is a list (I think for Android). I changed to be like this to keep some sort of standard. But LMK if I need to revert.

Comment (thymikee) [reply packages/cli/src/runIOS/runIOS.js]: Ok, that's good (name could be different now, but there's too much changes to this PR already :D) Thanks for bearing with me!

**Pull request merged**